Monthly newsletter for May (#251)
fortran-lang/fortran-lang.org: _posts/2021-05-01-Fortran-Newsletter-May-2021.md

---
layout: post
title: "Fortran newsletter: May 2021"
category: newsletter

> author: Sebastian Ehlert, Alexis Perry-Holby

Reviewer: ```suggestion
author: Sebastian Ehlert, Alexis Perry-Holby, Milan Curcic
```

    * Remove `%flang-new` from the LIT configuration
    * Update the regression tests to use the new driver when enabled
    * Add support for `-fget-definition`
* Move .f77 to the list of fixed-form file extensions
* Runtime
    * Implement reductions
    * Implement numeric intrinsic functions
    * TRANSFER() intrinsic function
    * RANDOM_NUMBER, RANDOM_SEED, RANDOM_INIT
    * Implement IPARITY, PARITY, and FINDLOC reductions
* Fix unit test failure on POWER
* Improve constant folding for type parameter inquiries
* Check for conflicting BIND(C) names
* Enforce a limit on recursive PDT instantiations
* Accept & fold IEEE_SELECTED_REAL_KIND
* Define missing & needed IEEE_ARITHMETIC symbols
* Handle instantiation of procedure pointer components
* Fix checking of argument passing for parameterized derived types
* Fix spurious errors from runtime derived type table construction
* Check for attributes specific to dummy arguments
* Handle structure constructors with forward references to PDTs

Call notes will be sent to the _flang-dev_ email list and also recorded [here](https://docs.google.com/document/d/10T-S2J3GrahpG4Ooif93NSTz2zBW0MQc_RlwHi0-afY).

### LFortran

> 

Reviewer 2: * 98 Merge Requests merged in April
* Working towards compiling the [SNAP](https://github.com/lanl/SNAP) proxy app ([#313](https://gitlab.com/lfortran/lfortran/-/issues/313)):
  * Code can be parsed to AST and transformed back to source code which compiles with other compilers and works
  * About 3rd of the files can be transformed from AST to ASR and the modules saved
* Other improvements:
  * Runtime library (more functions work)
  * Nested functions
  * Derived types

## Events

* We had our 11th Fortran Monthly call on April 22.
You can watch the recording below:

<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/D107yFcuZoE" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

> Wadud Miah ([@wadudmiah](https://github.com/wadudmiah)) from the University of Southampton presented a webinar on Fortran for High Performance Computing, organized by [Excellerat](https://www.excellerat.eu/). You can find the slides and the recording [here](https://services.excellerat.eu/viewevent/39).

Reviewer: ```suggestion
* Wadud Miah ([@wadudmiah](https://github.com/wadudmiah)) from the University of Southampton presented a webinar on Fortran for High Performance Computing, organized by [Excellerat](https://www.excellerat.eu/). You can find the slides and the recording [here](https://services.excellerat.eu/viewevent/39).
```

    * TRANSFER() intrinsic function
    * RANDOM_NUMBER, RANDOM_SEED, RANDOM_INIT
    * Implement IPARITY, PARITY, and FINDLOC reductions
* Fix unit test failure on POWER
* Improve constant folding for type parameter inquiries
* Check for conflicting BIND(C) names
* Enforce a limit on recursive PDT instantiations
* Accept & fold IEEE_SELECTED_REAL_KIND
* Define missing & needed IEEE_ARITHMETIC symbols
* Handle instantiation of procedure pointer components
* Fix checking of argument passing for parameterized derived types
* Fix spurious errors from runtime derived type table construction
* Check for attributes specific to dummy arguments
* Handle structure constructors with forward references to PDTs

Call notes will be sent to the _flang-dev_ email list and also recorded [here](https://docs.google.com/document/d/10T-S2J3GrahpG4Ooif93NSTz2zBW0MQc_RlwHi0-afY).

### LFortran

## Events

* We had our 11th Fortran Monthly call on April 22.
You can watch the recording below:

<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/D107yFcuZoE" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

> * Registration is open for the upcoming free webinar on
>   [Fortran for High Performance Computing](https://register.gotowebinar.com/register/7343048137688004108).
>   The webinar is organized by [Excellerat](https://www.excellerat.eu/)
>   and will be presented by Wadud Miah ([@wadudmiah](https://github.com/wadudmiah)) from the University of Southampton.

Reviewer: Edit with the link to slides and recording, the event has passed.

```suggestion
Wadud Miah ([@wadudmiah](https://github.com/wadudmiah)) from the University of Southampton presented a webinar on Fortran for High Performance Computing, organized by [Excellerat](https://www.excellerat.eu/). You can find the slides and the recording [here](https://services.excellerat.eu/viewevent/39).
```

> * Registration is open for the upcoming free webinar on
>   [Fortran for High Performance Computing](https://register.gotowebinar.com/register/7343048137688004108).
>   The webinar is organized by [Excellerat](https://www.excellerat.eu/)
>   and will be presented by Wadud Miah ([@wadudmiah](https://github.com/wadudmiah)) from the University of Southampton.
> 

Reviewer: This has passed so we should remove the note.

```suggestion
```

---
layout: post
title: "Fortran newsletter: May 2021"
category: newsletter

> author: Sebastian Ehlert, Alexis Perry-Holby, Milan Curcic

Reviewer 2: ```suggestion
author: Sebastian Ehlert, Alexis Perry-Holby, Milan Curcic,  Ondřej Čertík
```

    * Remove `%flang-new` from the LIT configuration
    * Update the regression tests to use the new driver when enabled
    * Add support for `-fget-definition`
* Move .f77 to the list of fixed-form file extensions
* Runtime
    * Implement reductions
    * Implement numeric intrinsic functions
    * TRANSFER() intrinsic function
    * RANDOM_NUMBER, RANDOM_SEED, RANDOM_INIT
    * Implement IPARITY, PARITY, and FINDLOC reductions
* Fix unit test failure on POWER
* Improve constant folding for type parameter inquiries
* Check for conflicting BIND(C) names
* Enforce a limit on recursive PDT instantiations
* Accept & fold IEEE_SELECTED_REAL_KIND
* Define missing & needed IEEE_ARITHMETIC symbols
* Handle instantiation of procedure pointer components
* Fix checking of argument passing for parameterized derived types
* Fix spurious errors from runtime derived type table construction
* Check for attributes specific to dummy arguments
* Handle structure constructors with forward references to PDTs

Call notes will be sent to the _flang-dev_ email list and also recorded [here](https://docs.google.com/document/d/10T-S2J3GrahpG4Ooif93NSTz2zBW0MQc_RlwHi0-afY).

> ### LFortran

Reviewer 2: ```suggestion
### LFortran

```
